cmd/kubelet/app/options/options.go

```diff
@@ -235,7 +235,8 @@ func ValidateKubeletFlags(f *KubeletFlags) error {
 
 	unknownLabels := sets.NewString()
 	for k := range f.NodeLabels {
-		if isKubernetesLabel(k) && !kubeletapis.IsKubeletLabel(k) {
+		openshiftMasterLabel := "node-role.kubernetes.io/master"
+		if isKubernetesLabel(k) && !(kubeletapis.IsKubeletLabel(k) || k == openshiftMasterLabel) {
```

Comment:
@sjenning @rphillips ptal

Reply:
Context: I added this change because 1.18 introduced testing of this function, and some of those tests were failing with the check for unknown labels [commented out](https://github.com/openshift/kubernetes/blob/origin-4.5-kubernetes-1.17.1/cmd/kubelet/app/options/options.go#L242).

Reply:
openshiftMasterLabel can be taken out of the for loop.... I suspect this change will prevent the Kubelet from starting since the node-role label gets added to the unknownLabels slice.